stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1550,7 +1551,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         _validateRange(subrange)
         let totalCount: Int = numericCast(newElements.count)
         _withStackOrHeapBuffer(totalCount) { conditionalBuffer in
-            let buffer = UnsafeMutableBufferPointer(start: conditionalBuffer.pointee.memory.assumingMemoryBound(to: UInt8.self), count: totalCount)
+            let buffer = UnsafeMutableBufferPointer(start: conditionalBuffer.pointee.memory.bindMemory(to: UInt8.self, capacity: totalCount), count: totalCount)
```

Comment:
That looks safe to me. It's harmless to bind newly allocated memory to whatever type is convenient for initialization. In general it would be good practice for `Data` to use only `UnsafeRawPointer` internally. For example, your could do this:
```
        let buffer = UnsafeMutableRawBufferPointer(start: conditionalBuffer.pointee.memory, count: totalCount)
        buffer.initializeMemory(as: UInt8.self, from: newElements)
```
It's effectively the same thing though, since `initializeMemory` implicitly binds the memory.

Reply:
The buffer in question here is conditionalBuffer which that is allocated from C and not part of Data.